### PR TITLE
chore(v3): get project config migration

### PR DESCRIPTION
**Pull request by jayzhang**
@microsoft/adaptivecards-tools	v1.0.3-rc.0
@microsoft/teamsfx-cli	v1.1.0-rc.0
@microsoft/teamsfx-core	v1.16.0-rc.2
@microsoft/teamsfx	v1.2.0-rc.1
@microsoft/teamsfx-react	v1.0.5-rc.1
ms-teams-vscode-extension	v4.0.6-rc.2

CLI BREAKING CHANGE:
  

Extension-toolkit BREAKING CHANGE:
 

CLI feat commits:
 feat: up fd24e4bd8
feat: up 02aab09c1 

Extension-toolkit feat commits:

### CODE REVIEW

**Comment by jayzhang** [comment]
hotfix: https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/15599927

File: packages/fx-core/src/core/FxCore.ts [hunk 1]
```diff
@@ -1070,7 +1073,9 @@ export class FxCore implements v3.ICore {
       if (result.isErr()) {
         return err(result.error);
       }
-      config.envInfos[env] = result.value;
+      const envInfo = result.value;
+      envInfo.state = convertEnvStateV3ToV2(envInfo.state) as v3.ResourceStates;
```
Comment (kuojianlu) [inline packages/fx-core/src/core/FxCore.ts:1077]: Why converting to V2 since the method is `getProjectConfigV3`?

Comment (jayzhang) [reply packages/fx-core/src/core/FxCore.ts:1077]: Ignore the name v3, this method is legacy implementation.

File: packages/fx-core/src/core/FxCore.ts [hunk 2]
```diff
@@ -1026,9 +1027,11 @@ export class FxCore implements v3.ICore {
     if (!ctx) return err(new ObjectIsUndefinedError("getProjectConfig input stuff"));
     inputs.stage = Stage.getProjectConfig;
     setCurrentStage(Stage.getProjectConfig);
+    let envState = ctx!.solutionContext?.envInfo.state;
+    if (envState) envState = convertEnvStateMapV3ToV2(envState);
     return ok({
       settings: ctx!.projectSettings,
-      config: ctx!.solutionContext?.envInfo.state,
+      config: envState,
```
Comment (kuojianlu) [inline packages/fx-core/src/core/FxCore.ts:1034]: Maybe need to check and ensure each call of this method uses the right key.

File: packages/fx-core/src/core/FxCore.ts (hunk 1 above)
Comment (kuojianlu) [reply packages/fx-core/src/core/FxCore.ts:1077]: This method may already be used with V3 keys, like here: https://github.com/OfficeDev/TeamsFx/blob/dev/packages/vscode-extension/src/debug/commonUtils.ts#L148.

File: packages/fx-core/src/core/FxCore.ts (hunk 2 above)
Comment (jayzhang) [reply packages/fx-core/src/core/FxCore.ts:1034]: FxCore makes sure the exported state and projectSettings are parity with v2.
I will unifty these two methods (getProjectConfig and getProjectConfigV3) in the future.

File: packages/fx-core/src/core/FxCore.ts (hunk 1 above)
Comment (jayzhang) [reply packages/fx-core/src/core/FxCore.ts:1077]: Oh, I see.

**Review by kuojianlu — APPROVED** [review]

File: packages/cli/src/cmds/preview/preview.ts [hunk 3]
```diff
@@ -782,9 +782,7 @@ export default class Preview extends YargsCommand {
       ?.get(constants.solutionPluginName)
       ?.get(constants.teamsAppTenantIdConfigKey) as string;
 
-    const remoteTeamsAppId: string = config?.config
-      ?.get(isV3() ? "app-manifest" : constants.appstudioPluginName)
-      ?.get(constants.remoteTeamsAppIdConfigKey);
+    const remoteTeamsAppId: string = config?.config?.get(constants.remoteTeamsAppIdConfigKey);
```
Comment (jayzhang) [inline packages/cli/src/cmds/preview/preview.ts:785]: FxCore makes sure that getProjectConfig follows the v2 plugin key rule.

**Review by kuojianlu — APPROVED** [review]

**Review by LongOddCode — APPROVED** [review]

**Pull request merged**